sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/BeamSqlDslAggregationTest.java

```diff
@@ -701,7 +700,6 @@ public void testSupportsAggregationWithoutProjection() throws Exception {
   }
 
   @Test
-  @Ignore("https://issues.apache.org/jira/browse/BEAM-8317")
   public void testSupportsAggregationWithFilterWithoutProjection() throws Exception {
```

Comment:
@11moon11 @apilloud 

What I really want to propose is when we add new test cases with SQL queries, run the test for both dialects unless there is a query syntax mismatch.

Using which planner is controlled by https://github.com/apache/beam/blob/master/sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/BeamSqlPipelineOptions.java#L28.


I am looking for a way of `@RunWith(Parameterized.class)` so it's easy to run tests for both dialect by an annotation.

Reply:
Found a useful reference link with examples: https://github.com/Pragmatists/JUnitParams/blob/master/src/test/java/junitparams/usage/SamplesOfUsageTest.java

Reply:
Cool. I agree this is a good idea, but we should hold off on doing this until #9737 is in. We will need to relocate common tests into another package.

Reply:
@apilloud 

I don't agree though. At least a duplicate test can be created but run for ZetaSQL only and then we can have a migration.

Reply:
I tried adding `pipeline.getOptions().as(BeamSqlPipelineOptions.class).setPlannerName("org.apache.beam.sdk.extensions.sql.zetasql.ZetaSQLPlannerImpl");` to the test to see if it will work using `ZetaSqlPlanner`, but I get `Class not found exception`.
I assume it is because ZetaSQL is not in the build file. After attempting to add a dependency there is the following error: `Circular dependency`, probably because ZetaSQL depends on BeamSQL?